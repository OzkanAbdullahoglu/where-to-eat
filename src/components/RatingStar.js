import React from 'react';

const RatingStar = ({ startStyle }) => (
  <li className={`cards-rating-star cards-rating-star${startStyle}`}></li>
);
export default RatingStar;
