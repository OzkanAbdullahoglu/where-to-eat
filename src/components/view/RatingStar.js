/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

const RatingStar = ({ startStyle }) => (
  <li className={`cards-rating-star cards-rating-star${startStyle}`}></li>
);

RatingStar.propTypes = {
  startStyle: PropTypes.string,
};

export default RatingStar;
