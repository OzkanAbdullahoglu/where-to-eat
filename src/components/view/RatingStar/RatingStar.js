/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

const RatingStar = ({ starStyle }) => (
  <li className={`cards-rating-star cards-rating-star${starStyle}`}></li>
);

RatingStar.propTypes = {
  starStyle: PropTypes.string,
};

export default RatingStar;
