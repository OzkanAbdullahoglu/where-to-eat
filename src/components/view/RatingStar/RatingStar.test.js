/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RatingStar from './RatingStar';

describe('Rating Star Component', () => {
  const props = {
    starStyle: '-half',
  };

  const ratingStarWrapper = shallow(<RatingStar {...props} />);
  it('renders Rating Star Component', () => {
    expect(toJson(ratingStarWrapper)).toMatchSnapshot();
  });
  it('renders `startStyle` props', () => {
    const startStyleProp = ratingStarWrapper.find('li').prop('className');
    expect(startStyleProp).toContain(props.starStyle);
  });
});
