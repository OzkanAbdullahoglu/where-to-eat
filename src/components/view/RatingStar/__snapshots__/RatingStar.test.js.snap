// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Star Component renders Rating Star Component 1`] = `
<li
  className="cards-rating-star cards-rating-star-half"
/>
`;
